lib/membrane_http_adaptive_stream/manifest/track.ex

```diff
@@ -70,15 +74,48 @@ defmodule Membrane.HTTPAdaptiveStream.Manifest.Track do
     and what elements are to be removed.
     """
 
-    @enforce_keys [:to_add, :to_remove]
-    defstruct @enforce_keys
+    defmodule Segment do
+      @type t :: %__MODULE__{
+              metadata: %{
+                duration: Membrane.Time.t() | Ratio.t(),
+                sequence_number: non_neg_integer()
+              },
+              name: String.t(),
+              payload: binary()
+            }
+      @enforce_keys [:metadata, :name, :payload]
+      defstruct @enforce_keys
+    end
 
-    @type element_type_t :: :header | :segment | :partial_segment
+    defmodule PartialSegment do
+      @type t :: %__MODULE__{
+              metadata: %{
+                duration: Membrane.Time.t() | Ration.t(),
+                sequence_number: non_neg_integer(),
+                independent?: boolean(),
+                byte_offset: non_neg_integer()
+              },
```

Comment:
We could have these fields in the struct directly, not under metadata. Also, consider having a single struct with the `type` field and some fields specific only to partial segments.

Reply:
I like the current approach, I'd only consider setting `byte_offset` to 0 and `independent?` to true for non-partial segments. But you can change it back to separate structs if you like. @Qizot wdyt>